### PR TITLE
Modify aug block in place

### PR DESCRIPTION
**Pull request by grishabh147**
Made some small changes to the code:
- Added functions to modify `AugmentedNLPBlockModel` in-place
- Made `fullspace.jl` compatible with MadNLP. Now `BlockNLPModels.jl` passes all tests with `MadNLP.jl` as the NLP solver.

### CODE REVIEW

File: src/augmented_block.jl [hunk 1]
```diff
@@ -121,6 +121,36 @@ function get_hessian(m::AugmentedNLPBlockModel;
     return H
 end
 
+"""
```
Comment (frapac) [inline src/augmented_block.jl:124]: that's exactly the idea.
I would just suggest a change in the name (for me, the primal and dual variables in the submodel are not solution, but parameters). How about just using `update_primal!` and `update_dual!`?

File: src/full_space.jl [hunk 2]
```diff
@@ -196,7 +196,7 @@ end
 
 function NLPModels.jac_coord!(nlp::FullSpaceModel, x::AbstractVector, vals::AbstractVector)
     @lencheck nlp.meta.nvar x
-    @lencheck nlp.meta.nnzj vals
+    # @lencheck nlp.meta.nnzj vals
```
Comment (frapac) [inline src/full_space.jl]: Maybe add a small comment to remember we comment out this line because of MadNLP?

File: test/simple_block_model.jl [hunk 3]
```diff
@@ -68,12 +68,14 @@ end
...
     @test hess(fm, x, y) ≈ tril(H(x, y))
+
+    # Solve the full-space problem with MadNLP
     stats = ipopt(fm, print_level = 0)
     @test stats.solution ≈ [0.41, 1, 1]
 
     # Solve the full-space problem with MadNLP
-    # To Do: Why MadNLP throws an error?
-    # madnlp(fm)
+    stats = madnlp(fm, print_level = MadNLP.WARN)
```
Comment (frapac) [inline test/simple_block_model.jl:77]: Nice to see `madnlp` is back!

**Review by frapac — APPROVED** [review]
Looks good to me! Ok to merge (I let you decide what to do with the naming issue)

**Comment by grishabh147** [comment]
@frapac, I think this can be merged now.

**Pull request merged**